### PR TITLE
Added in comprehensive docstring and added in minor changes

### PR DESCRIPTION
**Pull request by rayryeng**
Changes made:
- Added in a comprehensive docstring to document how to use the Neural Network engine.
- Made custom assert messages to shed light on what may be going wrong if the input is improper.  
- Changed default learning rate to 0.03 to reflect the default of the NeuralNetPlayground GUI.
- Fixed various minor typos in comments.

### CODE REVIEW

**Comment by amroamroamro** [comment]
merged, thanks Ray.
